AzureFunction/Request/TaskProperties.cs

```diff
@@ -0,0 +1,114 @@
+﻿namespace Microsoft.Azure.Pipelines.EvaluateArtifactPolicies.Request
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+
+    public class TaskProperties
+    {
+        public Guid ProjectId { get; }
+        public string HubName { get; }
+        public Guid PlanId { get; }
+        public string PlanUrl { get; }
+        public Guid JobId { get; }
+        public Guid TimelineId { get; }
+        public Guid TaskInstanceId { get; set; }
+        public string TaskInstanceName { get; }
+        public string AuthToken { get; }
+        public RequestType RequestType { get; }
+
+        private static readonly List<string> MandatoryProperties = new List<string>
+        {
+            ProjectIdKey,
+            PlanIdKey,
+            PlanUrlKey,
+            JobIdKey,
+            TimelineIdKey,
+            AuthTokenKey,
+        };
+
+        public static readonly List<string> PropertiesList = new List<string>(MandatoryProperties)
+        {
+            HubNameKey,
+            TaskInstanceIdKey,
+            TaskInstanceNameKey,
+            RequestTypeKey,
+        };
+
+        public TaskProperties(IDictionary<string, string> messageProperties)
+        {
+            var missingProperties = MandatoryProperties
+                .Where(propertyToCheck => !messageProperties.ContainsKey(propertyToCheck)).ToList();
+            if (missingProperties.Any())
+            {
+                var exceptionMessage =
+                    $"Required properties '{string.Join(", ", missingProperties)}' are missing. Please provide these values and try again.";
+                throw new InvalidDataException(exceptionMessage);
+            }
+
+            this.ProjectId = ParseGuid(messageProperties, ProjectIdKey);
+            this.JobId = ParseGuid(messageProperties, JobIdKey);
+            this.PlanId = ParseGuid(messageProperties, PlanIdKey);
+            this.TimelineId = ParseGuid(messageProperties, TimelineIdKey);
+
+            this.TaskInstanceId = messageProperties.ContainsKey(TaskInstanceIdKey)
+                ? ParseGuid(messageProperties, TaskInstanceIdKey)
+                : Guid.Empty;
+
+            this.HubName = "Build";
+            if (messageProperties.ContainsKey(HubNameKey))
+            {
+                this.HubName = messageProperties[HubNameKey];
```

Comment:
Simplify using `?` operator ?

Reply:
Can't use ? for non existent keys in a dictionary